src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -0,0 +1,418 @@
+
+const SelectorSubHeader = ({ children }) => div({ style: { fontSize: 12, fontWeight: 600 } }, children);
+
+interface HeaderAndValues<T extends DatasetBuilderType> {
+  header: string;
+  values: T[];
+}
+
+interface SelectorProps<T extends DatasetBuilderType> {
+  number: number;
+  header: string;
+  subheader?: string;
+  datasetBuilderObjectSets: HeaderAndValues<T>[];
+  selectedDatasetBuilderObjectSets: HeaderAndValues<T>[];
+  onChange: (newDatasetBuilderObjectSets: HeaderAndValues<T>[]) => void;
+  headerAction: any;
+  placeholder?: any;
+  style?: React.CSSProperties;
+}
+const Selector = <T extends DatasetBuilderType>({
+  number,
+  header,
+  subheader,
+  headerAction,
+  placeholder,
+  datasetBuilderObjectSets,
+  onChange,
+  selectedDatasetBuilderObjectSets,
+  style,
+}: SelectorProps<T>) => {
+  const selectedValues = _.flatMap(
+    (selectedDatasetBuilderObjectSet) =>
+      _.map(
+        (value) => ({ header: selectedDatasetBuilderObjectSet.header, value }),
+        selectedDatasetBuilderObjectSet.values
+      ),
+    selectedDatasetBuilderObjectSets
+  );
+  const isChecked = (datasetBuilderObjectSet, value) =>
+    _.intersectionWith(_.isEqual, [{ header: datasetBuilderObjectSet.header, value }], selectedValues).length > 0;
+
+  return li({ style: { width: '30%', ...style } }, [
+    div({ style: { display: 'flex', width: '100%', justifyContent: 'space-between', alignItems: 'flex-start' } }, [
+      div({ style: { display: 'flex' } }, [
+        div(
+          {
+            style: {
+              backgroundColor: colors.accent(0.2),
+              color: colors.dark(),
+              padding: '0.5rem',
+              borderRadius: '2rem',
+              fontSize: 14,
+              height: 24,
+              width: 24,
+              display: 'flex',
+              alignItems: 'center',
+              justifyContent: 'center',
+              fontWeight: 600,
+            },
+          },
+          [number]
+        ),
+        div({ style: { marginLeft: 10, height: '3rem' } }, [
+          h3({ style: { marginTop: 0, marginBottom: '0.5rem' } }, [header]),
+          div({ style: { fontSize: 12 } }, [subheader]),
+        ]),
+      ]),
+      headerAction,
+    ]),
+    div(
+      {
+        style: {
+          backgroundColor: 'white',
+          border: `1px solid ${colors.dark(0.5)}`,
+          borderRadius: 10,
+          height: 300,
+          padding: '1rem',
+          marginTop: '0.5rem',
+        },
+      },
+      [
+        datasetBuilderObjectSets &&
+        _.flatMap((datasetBuilderObjectSet) => datasetBuilderObjectSet.values, datasetBuilderObjectSets).length > 0
+          ? div(
+              { style: { display: 'flex', flexDirection: 'column', height: '100%', overflowY: 'auto' } },
+              _.map(
+                ([i, datasetBuilderObjectSet]) =>
+                  h(Fragment, [
+                    h(SelectorSubHeader, { key: `${datasetBuilderObjectSet.header}-${i}` }, [
```

Comment:
Is the index necessary for the key in this case? will any of the items in the list end up getting filtered / re-ordered?